src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -0,0 +1,352 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    public class CodeGenAsyncMainTests : EmitMetadataTestBase
+    {
+        [Fact]
+        public void GetResultReturnsSomethingElse()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+namespace System.Threading.Tasks {
+    public class Awaiter {
+        public double GetResult() { return 0.0; }
+    }
+    public class Task<T> {
+        public Awaiter GetAwaiter() {
+            return new Awaiter();
+        }
+    }
+}
+
+static class Program {
+    static Task<int> Main() {
+        return null;
+    }
+}";
+            var sourceCompilation = CreateCompilationWithMscorlib(source,  options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_1));
+            // Prototype: this should be a diagnostic.
+            Assert.Throws<System.Exception>(() => { sourceCompilation.VerifyEmitDiagnostics(); });
+        }
+
+        [Fact]
+        public void TaskOfTGetAwaiterReturnsVoid()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+namespace System.Threading.Tasks {
+    public class Task<T> {
+        public void GetAwaiter() {}
+    }
+}
+
+static class Program {
+    static Task<int> Main() {
+        return null;
+    }
+}";
+            var sourceCompilation = CreateCompilationWithMscorlib(source,  options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_1));
+            sourceCompilation.VerifyEmitDiagnostics(
+                // (12,12): warning CS0436: The type 'Task<T>' in '' conflicts with the imported type 'Task<TResult>' in 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'. Using the type defined in ''.
+                //     Task<int>
+                Diagnostic(ErrorCode.WRN_SameFullNameThisAggAgg, "Task<int>").WithArguments("", "System.Threading.Tasks.Task<T>", "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", "System.Threading.Tasks.Task<TResult>").WithLocation(12, 12),
+                // (12,5): error CS0656: Missing compiler required member 'void.GetResult'
```

Comment:
>// (12,5): error CS0656: Missing compiler required member 'void.GetResult' [](start = 16, length = 74)

This error is confusing, please add a comment to make sure it is changed when we fix lookup implementation.